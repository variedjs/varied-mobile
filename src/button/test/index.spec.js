import { mount } from "../../../test";
import Button from "..";

test("loading-size prop", () => {
  const wrapper = mount(Button, {
    propsData: {
      loading: true,
      loadingSize: "10px"
    }
  });
  expect(wrapper).toMatchSnapshot();
});

test("icon-position prop", () => {
  const wrapper = mount(Button, {
    propsData: {
      icon: "plus",
      iconPosition: "right"
    }
  });
  expect(wrapper).toMatchSnapshot();
});

test("click event", () => {
  const onClick = jest.fn();
  const wrapper = mount(Button, {
    listeners: {
      click: onClick
    }
  });

  wrapper.trigger("click");
  expect(onClick).toHaveBeenCalled();
});

test("not trigger click event when disabled", () => {
  const onClick = jest.fn();
  const wrapper = mount(Button, {
    propsData: {
      disabled: true
    },
    listeners: {
      click: onClick
    }
  });

  wrapper.trigger("click");
  expect(onClick).toHaveBeenCalledTimes(0);
});

test("not trigger click event when loading", () => {
  const onClick = jest.fn();
  const wrapper = mount(Button, {
    propsData: {
      loading: true
    },
    listeners: {
      click: onClick
    }
  });

  wrapper.trigger("click");
  expect(onClick).toHaveBeenCalledTimes(0);
});

// test("touchstart event", () => {
//   const onTouchstart = jest.fn();
//   const wrapper = mount(Button, {
//     listeners: {
//       touchstart: onTouchstart
//     }
//   });
//
//   wrapper.trigger("touchstart");
//   expect(onTouchstart).toHaveBeenCalled();
// });

test("icon-prefix prop", () => {
  const wrapper = mount(Button, {
    propsData: {
      icon: "success",
      iconPrefix: "my-icon"
    }
  });

  expect(wrapper).toMatchSnapshot();
});

test("loading slot", () => {
  const wrapper = mount(Button, {
    propsData: {
      loading: true
    },
    scopedSlots: {
      loading: () => "Custom Loading"
    }
  });

  expect(wrapper).toMatchSnapshot();
});
