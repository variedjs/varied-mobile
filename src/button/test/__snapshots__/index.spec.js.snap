// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon-position prop 1`] = `
<button class="vm-button vm-button-md vm-button-nc">
  <!----> <i class="vm-icon vm-icon-plus vm-button-icon">
    <!---->
  </i>
</button>
`;

exports[`icon-prefix prop 1`] = `
<button class="vm-button vm-button-md vm-button-nc" iconprefix="my-icon"><i class="vm-icon vm-icon-success vm-button-icon">
    <!---->
  </i>
  <!---->
</button>
`;

exports[`loading slot 1`] = `
<button class="vm-button vm-button-loading vm-button-md vm-button-nc"><i class="vm-icon vm-button-icon">
    <!---->
  </i> <i class="vm-icon-spin vm-icon vm-icon-spinner">
    <!---->
  </i>
  <!---->
</button>
`;

exports[`loading-size prop 1`] = `
<button class="vm-button vm-button-loading vm-button-md vm-button-nc" loadingsize="10px"><i class="vm-icon vm-button-icon">
    <!---->
  </i> <i class="vm-icon-spin vm-icon vm-icon-spinner">
    <!---->
  </i>
  <!---->
</button>
`;
